src/main/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/service/GroupsService.kt

```diff
@@ -90,19 +95,51 @@ class GroupsService(
       null
     )
   }
-}
 
-class GroupNotFoundException(action: String, group: String, errorCode: String) :
-  Exception("Unable to $action group: $group with reason: $errorCode")
+  @Transactional
+  @Throws(GroupNotFoundException::class, GroupHasChildGroupException::class)
+  fun deleteGroup(groupCode: String) {
+    val group =
+      groupRepository.findByGroupCode(groupCode) ?: throw GroupNotFoundException("delete", groupCode, "notfound")
+
+    when {
+      group.children.isEmpty() -> {
+        removeUsersFromGroup(
+          groupCode,
+          authenticationFacade.currentUsername,
+          authenticationFacade.authentication.authorities
+        )
+        groupRepository.delete(group)
+
+        telemetryClient.trackEvent(
+          "GroupDeleteSuccess",
+          mapOf("username" to authenticationFacade.currentUsername, "groupCode" to groupCode),
+          null
+        )
+      }
+      else -> {
+        throw GroupHasChildGroupException(groupCode, "child group exist")
+      }
+    }
+  }
 
```

Comment:
Better if all exceptions are not nested?

Reply:
agreed ..will move it out as our standard